shared/reducers/search.js

```diff
@@ -12,6 +12,7 @@ export type State = {
   searchText: ?string,
   searchIcon: IconType,
   searchPlatform: SearchPlatforms,
+  searchTextClearTrigger: number,
```

Comment:
Add a trigger which is driven by redux. if this thing changes redux is telling us to blow away our search input